admin/class-boldgrid-backup-admin-task.php

```diff
@@ -0,0 +1,292 @@
+				( empty( $this->date_format ) ? $this->completed_at : date( $this->date_format, $this->completed_at ) ),
+			'status'       => $this->get_status(),
+			'data'         => $this->data,
+		];
+	}
+
+	/**
+	 * Get a specific value from the data.
+	 *
+	 * @since SINCEVERSION
+	 *
+	 * @param  string $key     The key to get.
+	 * @param  mixed  $default The default value to return.
+	 * @return mixed
+	 */
+	public function get_data( $key, $default = false ) {
+		return isset( $this->data[ $key ] ) ? $this->data[ $key ] : $default;
+	}
+
+	/**
+	 * Get our task id.
+	 *
+	 * @since SINCEVERSION
+	 *
+	 * @return string
+	 */
+	public function get_id() {
+		return $this->id;
+	}
+
+	/**
+	 * Get the status of this task.
+	 *
+	 * @since SINCEVERSION
+	 *
+	 * @return string
+	 */
+	public function get_status() {
+		if ( empty( $this->started_at ) ) {
+			$status = 'pending';
+		} elseif ( empty( $this->completed_at ) ) {
+			$status = 'in_progress';
+		} else {
+			$status = 'done';
+		}
+
+		return $status;
+	}
+
+	/**
+	 * Initialize a new task.
+	 *
+	 * @since SINCEVERSION
+	 *
+	 * @param  array $args An array of args.
+	 * @return bool        True if this task was initialized.
+	 */
+	public function init( $args ) {
+		$this->id           = ! empty( $args['id'] ) ? $args['id'] : $this->set_id();
+		$this->created_at   = ! empty( $args['created_at'] ) ? $args['created_at'] : time();
+		$this->started_at   = ! empty( $args['started_at'] ) ? $args['started_at'] : null;
+		$this->completed_at = ! empty( $args['completed_at'] ) ? $args['completed_at'] : null;
+		$this->data         = ! empty( $args['data'] ) ? $args['data'] : [];
+
+		// You must supply a type.
+		$type = ! empty( $args['type'] ) ? $args['type'] : null;
+		if ( empty( $type ) ) {
+			return false;
+		} elseif ( ! $this->set_type( $type ) ) {
+			return false;
+		}
+
+		return true;
+	}
+
+	/**
+	 * Init this task by a task id.
+	 *
+	 * @since SINCEVERSION
+	 *
+	 * @param  string $id A task id.
+	 * @return bool       True if this task was initialized.
+	 */
+	public function init_by_id( $id ) {
+		$task = $this->helper->get_by_id( $id );
+
+		return $this->init( $task );
+	}
+
+
+	/**
+	 * Create a new task id.
+	 *
+	 * @since SINCEVERSION
+	 *
+	 * @return string
+	 */
+	private function set_id() {
+		// A task id is the current time + 6 random chars.
+		$this->id = time() . '-' . substr( md5( time() ), -6 );
+
+		return $this->id;
+	}
+
+	/**
+	 * Set our task type.
+	 *
+	 * @since SINCEVERSION
+	 *
+	 * @param  string $type Our task type.
+	 * @return bool         True if the type was set.
+	 */
+	private function set_type( $type ) {
+		$valid_types = [
+			'backup',
+			'restore',
+		];
+
+		if ( ! in_array( $type, $valid_types, true ) ) {
+			return false;
+		}
+
+		$this->type = $type;
+
+		return true;
+	}
+
+	/**
+	 * Start this task.
+	 *
+	 * @since SINCEVERSION
+	 */
+	public function start() {
+		$this->started_at = time();
+
+		$this->update();
+	}
+
+	/**
+	 *
```

Comment:
This docblock is empty

Reply:
https://github.com/BoldGrid/boldgrid-backup/commit/67b4e8be4f61fb11b8e6fe3de7b18f4868072297